plugins/search/providers/elastic_search/client/elasticClient.php

```diff
@@ -462,6 +462,18 @@ public function ping()
 		$response = $this->sendRequest($cmd, self::GET, null, false, self::ELASTIC_ACTION_PING, self::MONITOR_NO_INDEX);
 		return $response;
 	}
+	
+	/**
+	 * get elastic cluster name
+	 * @return mixed|string
+	 */
+	public function getElasticClusterName()
+	{
+		$response = $this->ping();
+		
+		// this is backward compatible with old method: $elasticClusterName = kConf::get('elasticCluster', 'elastic', 0);
+		return isset($response['cluster_name']) ? $response['cluster_name'] : 0;
```

Comment:
I think:
1. If cluster_name is not set, it means [elastic_host]:9200 did not respond - so not having 'cluster_name' would be the least of our concerns in which point we probably will switch to backup cluster which will then fetch the cluster_name immediately as it's not stored in cache
2. I added a retry of 3 times with sleep of 1 sec, just in case.
3. we can either:
* write cluster_name value to memcache with no ttl, if retry fails 3 times - get the backup value.
* the backup value will only be changed when value != backup value so we always keep the 'latest'
* get from kConf - which will be the 'easiest' but more chances of being less accurate.

wdyt?

Reply:
@yossipapi
1. What did we decide for backup value ? Do we get if from kConf / backup memcache (you had another idea that slipped my mind)
2. Do we want to remove the retry? I prefer system integrity (at cost of keeping the apache up to 3 seconds) over performance at cost of system integrity.

Reply:
Currently we resolved it by adding retry.